server/src/code-runner/docker-runner.ts

```diff
@@ -91,16 +112,9 @@ export class DockerRunner implements CodeRunner {
               })
             ),
             concat(this.config.downloader.fetch(containerId, configuration.inputs)),
-            concat(
-              this.config.spawn(this.config.dockerExecutable, [
-                'exec',
-                '-t',
-                containerId,
-                '/bin/bash',
-                '-c',
-                `mkdir /run/outputs && cd /run && python main.py ${args}`
-              ])
-            ),
+            // Only run the `preExecutionCommand` if the container is writeable
+            concat(this.runCommand(containerId, `cd /run && ${configuration.preExecutionCommand}`, environment.writeable)),
```

Comment:
If the default  for the `preExecutionCommand` is an empty string, won't that yield an error message by default?

Reply:
Good catch! Will fix it.

Reply:
Probably best to trim and ignore empty entirely